File: python/leap/leap.py


def _evenly_divisible(n, m):
    return not n % m


def is_leap_year(year):
    if not _evenly_divisible(year, 4):
        return False
    elif _evenly_divisible(year, 100):
        return _evenly_divisible(year, 400)
    else:
        return True

File: go/leap/leap.go
package leap

const testVersion = 3

// IsLeapYear determines if a year is a leap year
func IsLeapYear(year int) bool {
	return multiple(year, 400) || multiple(year, 4) && !multiple(year, 100)
}

func multiple(n, m int) bool {
	return n%m == 0
}
